client/signup/hooks/use-branch-steps.ts

```diff
@@ -1,31 +1,33 @@
 import React from 'react';
-import { useDispatch } from 'react-redux';
+import { useSelector, useDispatch } from 'react-redux';
 import flows from 'calypso/signup/config/flows';
 import { addExcludeSteps, removeExcludeSteps } from 'calypso/state/signup/flow/actions';
+import { getSignupProgress } from 'calypso/state/signup/progress/selectors';
 
-type BranchSteps = ( excludeSteps: string[] ) => void;
+type GetExcludeSteps< Values > = ( values?: Values ) => string[];
```

Comment:
I think, we don't need this parameterized type `Values`. To better convey the intention, can we reuse `Dependencies` type here?

https://github.com/Automattic/wp-calypso/blob/595a1a7cc7a7031766c8751f59291be4bc17477a/client/signup/types.ts#L1-L3

(and also renaming `values` => `providedDependencies`)

Reply:
I think the `Dependencies` here is weird as the value might be a number, string, and so on rather than `string[]` so I prefer to let each step define what kind of values it provides.

But renaming to `providedDependencies` is great as I forgot `submitSignupStep` also call it `providedDependencies` 👍

Reply:
To be clearer, I was proposing something like this:

```
type GetExcludedSteps = ( providedDependencies?: Dependencies ) => string[];
type BranchSteps = ( providedDependencies: Dependencies ) => void;
```

Reply:
Ah I think we commented at the same time haha. Okay, let me check it again... 😄

Reply:
Hmmm...I think fixing the type might be better and make it simple

Reply:
Okay I think I now understand what's happening here....

Still I think introducing a parameterized type `Values` here does not really add any value, as the clients are using `any` anyway:

```
const getExcludeSteps = ( values: any ) => EXCLUDE_STEPS[ values?.intent ];
                                   ^
                                  here
```

so we won't get benefit from the type system. Could you check/confirm this?

Reply:
> Hmmm...I think fixing the type might be better and make it simple

Race condition again, commenting at the same time 😂 

Sure sure, could you figure out what's the best action for this PR 👍